google-userinfo/background/main.js

```diff
@@ -0,0 +1,23 @@
+
+function notifyUser(user) {
+  browser.notifications.create({
+    "type": "basic",
+    "title": "Google info",
+    "message": `Hi ${user.name}`
+  });}
+
+function logError(error) {
+  console.error(`Error: ${error}`);
+}
+
+/*
+When the button's clicked:
+- get an access token using the identity API
+- use it to get the user's info
+- show a notification containing some of it
+*/
+browser.browserAction.onClicked.addListener(() => {
+  getAccessToken().then(getUserInfo)
+                  .then(notifyUser)
+                  .catch(logError);
```

Comment:
Making the indentation depend on the length of the method `getAccessToken()` is a real pain when you change the code later and use a different API then have to re-indent all the lines below it.

We ran into this so much in python that we just indent at a fixed limit like Shane suggests.